app/views/tasks/_rating_modal.html.slim

```diff
@@ -1,27 +1,17 @@
-.modal.fade#ratingModal aria-hidden="true" data-task-id=@task.id tabindex="-1"
-  .modal-dialog.modal-dialog-centered.modal-sm
-    .modal-content
-      .modal-header
-        h6.modal-title
-          = Rating.model_name.human
-        button.btn-close data-bs-dismiss="modal" aria-label="Close"
-      .modal-body
-        - Rating::SUBCATEGORIES.each do |category|
-          span.rating-category-container
-            span
-              = t("activerecord.attributes.rating.#{category}")
-            span.task-rating
-              = render('task_star_rating', rating: @user_rating.send(category), is_input: true, category:)
-        hr
-        span.rating-category-container
-          span
-            = t('activerecord.attributes.rating.overall_rating')
-          span.task-rating
-            = render('task_star_rating', rating: @user_rating.overall_rating, is_input: true, category: :overall_rating)
-      .modal-footer.pt-0.border-0
-        - if @user_rating.valid?
-          button.btn.btn-light#ratingModalSaveButton data-bs-dismiss="modal"
-            = t('common.button.save')
-        - else
-          button.btn.btn-light#ratingModalSaveButton[data-bs-dismiss="modal" disabled]
-            = t('common.button.save')
+= render 'shared/modal', title: Rating.model_name.human, root_attributes: {id: 'ratingModal', 'data-task-id': @task.id}, size: 'modal-sm', footer: false do
+  .modal-body
```

Comment:
I would prefer to move this `div` over to `app/views/shared/_modal.html.slim`

Reply:
I know but if we do that we cannot use the modal template with a custom modal-header or modal-footer because everything gets wrapped in the modal-body. Ideally we could pass 3 blocks to the modal partial - one for header, body and footer - but ruby allows only one block. I could not get it to work like that so I went with this instead to allow custom headers/footers.

Reply:
Okay, so I used `content_for` now to achieve something like that. I think this is much better. There is a problem with content_for though: If you define the same `content_for` twice the default behavior is to append the contents which is not what we want if we define multiple modals on one page. There is a flush option but still: If one modal defines a custom footer through content_for and another modal wants to use the default footer this might conflict (if we use `content_for?` to check if we want to include a custom footer) so I had to add two additional flags `custom_header` and `custom_footer` to allow the second modal to ignore the defined custom footer of the first modal.

Reply:
We included the `clear_content_for` method and cleaned the modal template a little before merging.